src/tools/rbd/action/Wnbd.cc

```diff
@@ -81,6 +81,9 @@ int get_image_or_snap_spec(const po::variables_map &vm, std::string *spec) {
   if (!pool_name.empty()) {
     spec->append(pool_name);
     spec->append("/");
+  } else {
+    spec->append(utils::get_default_pool_name());
```

Comment:
Calling `get_default_pool_name` without connecting the cluster first is a problem because the desired `rbd_default_pool` value may be stored in the centralized config.  What was the reason for adding your own `get_image_or_snap_spec` in `Wnbd.cc` instead of reusing one from `Nbd.cc` which does the right thing?

Reply:
Good point. We did reuse the function, it's just that it was copied as we weren't sure if calling functions from Nbd.cc would be safe (e.g. it might contain linux or nbd specific bits at some point) and saw that Ggate.cc also had a copy . The pool fix was added later: https://github.com/ceph/ceph/commit/987e017f20dc419f45f222a35c9ccca79551af7d

Reply:
Let's pull it into `src/tools/rbd/Utils.cc` to avoid something like this happening again in the future.

Reply:
both `get_image_or_snap_spec` and `parse_options` have been moved to `Utils.cc` and the copies in `Wnbd.cc`, `Nbd.cc` and `Ggate.cc` have been dropped